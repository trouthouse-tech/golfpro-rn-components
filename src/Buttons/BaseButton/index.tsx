import React from 'react';
import { TouchableHighlight, StyleSheet } from 'react-native';

type BaseButtonProps = {
  onPress(): void;
  buttonStyle: {};
  child: React.ReactNode;
}

export default function BaseButton(props: BaseButtonProps) {
  const { onPress, buttonStyle, child } = props;
  return (
    <TouchableHighlight
      style={[styles.button, buttonStyle]}
      onPress={() => onPress()}
      testID="basebutton">
      {child}
    </TouchableHighlight>
  );
}

const styles = StyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
  },
});
