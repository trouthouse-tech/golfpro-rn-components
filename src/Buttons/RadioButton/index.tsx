import { View, StyleSheet, Text, TouchableHighlight } from 'react-native';
import React from 'react';
import { Fonts } from '../../styles/';

type RadioButtonProps = {
  isActive: boolean;
  text: string;
  onPress(): void;
};

export default function RadioButton({ isActive, text, onPress }: RadioButtonProps) {
  return (
    <TouchableHighlight onPress={() => onPress()}>
      <View style={styles.button}>
        <View style={styles.outside}>
          {isActive ? <View style={styles.inside} /> : null}
        </View>
        <Text style={styles.text}>{text}</Text>
      </View>
    </TouchableHighlight>
  );
}

const styles = StyleSheet.create({
  button: {
    flexDirection: 'row',
  },

  outside: {
    height: 24,
    width: 24,
    borderRadius: 12,
    borderWidth: 2,
    borderColor: '#000',
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: 20,
  },

  inside: {
    height: 12,
    width: 12,
    borderRadius: 6,
    backgroundColor: '#000',
  },

  text: {
    marginLeft: 5,
    fontSize: Fonts.normal,
  },
});
