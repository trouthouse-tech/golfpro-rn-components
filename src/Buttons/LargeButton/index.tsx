import React from 'react';
import { StyleSheet, Text } from 'react-native';
import { Fonts, Utils, Colors } from '../../styles/';
import BaseButton from '../BaseButton';

type LargeButtonProps = {
  onPress(): void;
  text: string;
  buttonColor?: string;
  textColor?: string;
  borderColor?: string;
  underlayColor?: string;
  disabled?: boolean;
}

export default function LargeButton(props: LargeButtonProps) {

  const {
    onPress,
    text,
    buttonColor,
    textColor,
    borderColor,
    underlayColor,
    disabled,
  } = props;

  const child = (
    <Text style={[styles.buttonText, { color: textColor || Colors.white }]}>
      {text}
    </Text>
  );

  const customStyles = {
    backgroundColor: buttonColor || Colors.main_green,
    borderColor: borderColor || Colors.main_green,
    borderWidth: 1,
  };

  return (
    <BaseButton
      onPress={() => onPress()}
      buttonStyle={[styles.largeButton, customStyles]}
      child={child}
      underlayColor={underlayColor || buttonColor || Colors.main_green}
      disabled={disabled}
    />
  );
}

const styles = StyleSheet.create({
  largeButton: {
    width: Utils.DEVICE_WIDTH / 1.2,
    height: Utils.DEVICE_HEIGHT / 12,
    borderRadius: 5,
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 10,
  },

  buttonText: {
    fontSize: Fonts.normal,
    fontWeight: 'bold',
  },
});
