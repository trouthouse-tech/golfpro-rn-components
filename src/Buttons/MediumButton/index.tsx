import React from 'react';
import {StyleSheet, Text} from 'react-native';
import {Fonts, Utils, Colors} from '../../styles';
import BaseButton from '../BaseButton';
import {ButtonProps} from '../';

export default function MediumButton({
  onPress,
  text,
  buttonColor,
  textColor,
  borderColor,
}: ButtonProps) {
  const child = (
    <Text style={[styles.buttonText, {color: textColor || Colors.white}]}>
      {text}
    </Text>
  );
  const customStyles = {
    backgroundColor: buttonColor || Colors.main_green,
    borderColor: borderColor || Colors.main_green,
    borderWidth: 1,
  };

  return (
    <BaseButton
      onPress={() => onPress()}
      buttonStyle={[styles.mediumSquare, customStyles]}
      child={child}
    />
  );
}

const styles = StyleSheet.create({
  mediumSquare: {
    width: Utils.DEVICE_WIDTH / 1.7,
    height: Utils.DEVICE_HEIGHT / 12,
    borderRadius: 5,
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 10,
  },

  buttonText: {
    fontSize: Fonts.normal,
    fontWeight: 'bold',
  },

  button: {
    width: Utils.DEVICE_WIDTH / 1.2,
    height: Utils.DEVICE_HEIGHT / 15,
    borderRadius: 10,
  },
});
