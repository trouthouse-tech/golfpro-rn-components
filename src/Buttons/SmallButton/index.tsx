import React from 'react';
import { StyleSheet, Text } from 'react-native';
import { Fonts, Utils, Colors } from '../../styles/';
import BaseButton from '../BaseButton';

type SmallButtonProps = {
  onPress(): void;
  text: string;
  buttonColor?: string;
  textColor?: string;
  borderColor?: string;
  additionalStyling?: {};
}

export default function SmallButton(props: SmallButtonProps) {

  const {
    onPress,
    text,
    buttonColor,
    textColor,
    borderColor,
    additionalStyling,
  } = props;

  const child = (
    <Text style={[styles.buttonText, { color: textColor || Colors.main_green }]}>
      {text}
    </Text>
  );
  const customStyles = {
    backgroundColor: buttonColor || Colors.main_green,
    borderColor: borderColor || Colors.main_green,
    borderWidth: 1,
  };

  return (
    <BaseButton
      onPress={() => onPress()}
      buttonStyle={[styles.small, customStyles, additionalStyling]}
      child={child}
    />
  );
}

const styles = StyleSheet.create({
  small: {
    width: Utils.DEVICE_WIDTH / 3,
    height: Utils.DEVICE_HEIGHT / 12,
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 10,
  },

  buttonText: {
    fontSize: Fonts.normal,
    fontWeight: 'bold',
  },

  button: {
    width: Utils.DEVICE_WIDTH / 1.2,
    height: Utils.DEVICE_HEIGHT / 15,
    borderRadius: 10,
  },
});
