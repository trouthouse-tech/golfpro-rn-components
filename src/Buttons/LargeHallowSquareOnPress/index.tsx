import React from 'react';
import LargeButton from '../LargeButton';
import {Colors} from '../../../styles';
import {ButtonProps} from '../types';

export default function LargeHallowSquareOnPress(props: ButtonProps) {
  return <LargeButton {...props} buttonColor={Colors.white} />;
}
