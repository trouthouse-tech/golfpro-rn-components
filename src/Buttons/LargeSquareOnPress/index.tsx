import React from 'react';
import LargeButton from '../LargeButton';
import {ButtonProps} from '../';

export default function LargeSquareOnPress(props: ButtonProps) {
  return <LargeButton borderColor={props.buttonColor} {...props} />;
}
