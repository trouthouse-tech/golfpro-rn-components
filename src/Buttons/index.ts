import LargeHallowSquareOnPress from './LargeHallowSquareOnPress';
import LargeButton from './LargeButton';
import LargeSquareOnPress from './LargeSquareOnPress';
import TextButton from './TextButton';
import SmallHallowButton from './SmallHallow';
import SmallHallowRoundedButton from './SmallHallowRoundedButton';
import MediumButton from './MediumButton';
import SmallButton from './SmallButton';

export {
  LargeHallowSquareOnPress,
  LargeButton,
  LargeSquareOnPress,
  TextButton,
  SmallButton,
  SmallHallowButton,
  SmallHallowRoundedButton,
  MediumButton,
};
