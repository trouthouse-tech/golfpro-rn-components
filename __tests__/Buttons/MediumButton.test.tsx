import React from 'react';
import { MediumButton } from "../../src/Buttons";

import { render, cleanup, fireEvent } from "@testing-library/react-native";
import '@testing-library/jest-native/extend-expect';
import renderer from 'react-test-renderer';

describe('MediumButton', () => {

  const mockClickFunction = jest.fn();

  afterEach(cleanup);
  it('renders and matches snapshot', () => {
    const tree = renderer.create(<MediumButton text="Request Lesson" onPress={mockClickFunction} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders a clickable MediumButton", () => {
    const { getByText } = render(
      <MediumButton text="Request Lesson" onPress={mockClickFunction} />
    );

    const button = getByText("Request Lesson");
    fireEvent.press(button); //press the button
    expect(mockClickFunction).toHaveBeenCalledTimes(1);
  });

  it("styles MediumButton correctly", () => {
    const { getByTestId } = render(
      <MediumButton text="Request Lesson" onPress={mockClickFunction}
        buttonColor="red" borderColor="red" />
    );

    const baseButton = getByTestId("basebutton");
    expect(baseButton).toHaveStyle({
      backgroundColor: 'red',
      borderColor: 'red'
    });
  });

});

