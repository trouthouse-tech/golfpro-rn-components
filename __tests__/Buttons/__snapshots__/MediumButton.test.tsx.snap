// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediumButton renders and matches snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 5,
          "height": 111.16666666666667,
          "justifyContent": "center",
          "marginTop": 10,
          "width": 441.1764705882353,
        },
        Object {
          "backgroundColor": "#009b29",
          "borderColor": "#009b29",
          "borderWidth": 1,
        },
      ],
    ]
  }
  testID="basebutton"
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
        },
        Object {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    Request Lesson
  </Text>
</View>
`;
